### ISSUE TITLE
kubectl desribe node always return the same machine ID

### ISSUE THREAD

**Issue by crazybber** [rancher/rancher#16236]
rancher/rancher:v2.1.1 

Steps to reproduce:

1. Create a cluster with custom or node-template
2. Use `kubectl describe node` to find the machine ID

Results:

It always return `8c859f2bde9e41155bca40d2d85aebb5 `

**Comment by loganhz** [comment]
Can you supply more info on your setup (basically copy paste from this and fill in: https://raw.githubusercontent.com/rancher/rancher/master/.github/ISSUE_TEMPLATE.md)

**Comment by loganhz** [comment]
FYI https://www.freedesktop.org/software/systemd/man/machine-id.html

**Comment by crazybber** [comment]
@loganhz not a difficult process .

just check some different single node environment . by kubectl  . 

step1 : kubectl describe node
step2: check the machineID.

**Comment by crazybber** [comment]
@loganhz http://man7.org/linux/man-pages/man5/machine-id.5.html 

different  should have different machineID whatever hashed or not.

**Comment by loganhz** [comment]
How did you add this cluster? I guess you add it as a custom cluster.
So you already have your machine ready with OS and docker.

The machine ID is assigned when you install the OS. 
Rancher and Kubernetes can not change it..

**Comment by crazybber** [comment]
@loganhz you didn't get what i mean.

i have two bare-metal  machine .with different machineID. 
machine 1  machine ID : 123456
machine 2  machine ID : ABCDEF

but when i queryed the machineID in rancher cluster by kubectl  i just get a fixed string: 8c859f2bde9e41155bca40d2d85aebb5 ,not 123456 or ABCDEF.

**Comment by crazybber** [comment]
![image](https://user-images.githubusercontent.com/3401462/47340228-1fcefa80-d6d0-11e8-8c45-ac38922aba01.png)
![image](https://user-images.githubusercontent.com/3401462/47340276-48ef8b00-d6d0-11e8-8f94-a7197aab0cd4.png)

**Comment by loganhz** [comment]
Thanks ! I got it

**Comment by crazybber** [comment]
OK, if you got a work-round , or quick fix ,please let me know ,here, i will keep watch.

**Comment by crazybber** [comment]
how it goes ?

**Comment by loganhz** [comment]
I guess it will be fixed in 2.2. I guess you can work around it by mounting `/etc/machine-id` to kubelet.https://rancher.com/docs/rke/v0.1.x/en/config-options/services/services-extras/

**Comment by cjellick** [comment]
@crazybber can you describe the impact of this bug? Like, what does it break for you? Will help me in assessing impact and assigning priority

**Comment by erikwilson** [comment]
We may want to fix this in the hyperkube image

**Comment by crazybber** [comment]
@cjellick we need get the uuid of the node  to do some critical business  control.  in which case ,the uuid should change  easily , boot ID and System UUID are not  proper.

we followed the :https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#nodesysteminfo-v1-core 
MachineID reported by the node. For unique machine identification in the cluster this field is preferred.

**Comment by erikwilson** [comment]
I think that we should make machine-id a first-class configuration option for RKE, with some caveats.

My first concern is that machine-id might be used for encryption or cryptography purposes, which I would argue is a very bad idea, for this very issue. The portability of the app is limited to the implementation of the kubernetes provider, and there is no guarantee of randomness. On Docker for Mac there is no machine-id. There is no promise about how much information cloud providers may leak for network accessibility or how non-random those IDs may be. There are two phrases that jumped out at me in the linked documentation: `This ID uniquely identifies the host. It should be considered "confidential", and must not be exposed in untrusted environments, in particular on the network.`, and `The machine ID is usually generated from a random source`. I believe the confidential part is obsolete, and contrast that to Debian documentation from where the machine-id is actually generated: https://manpages.debian.org/stretch/systemd/machine-id.5.en.html.

**Comment by crazybber** [comment]
@erikwilson agreed with u, better to make marchine-id  default to the system machine-id ,but configurable in the RKE config, in which it can be replaced with a custom value.

**Comment by crazybber** [comment]
hi,guys,six months passed,should it take so long to handle this issue ?

**Comment by crazybber** [comment]
did we have any update ?
